zulipterminal/ui_tools/buttons.py

```diff
@@ -164,7 +164,7 @@ def __init__(self, properties: List[Any],
 
         # Mark muted streams 'M' during button creation.
         if self.model.is_muted_stream(self.stream_id):
-            self.mark_muted()
+            self.update_widget('M')
```

Comment:
Did the model changes expose something different here?

Looking at the unread counts handling being here, I'd love to move that into a model method to mark a stream/topic muted; it's rather hacky being in UI code.

Reply:
Yeah, that seems like a useful refactor too.

Reply:
Unrelated, so opened #722